flake.nix

```diff
@@ -2,9 +2,7 @@
   description = "hs-backend-booster";
 
   inputs = {
-    k-framework.url = "github:runtimeverification/k/v5.6.131";
```

Comment:
The changes to the nix flake are all to do with removing a direct dependence on the k-framework inside the flake. This is due to the fact that depending on the k-framework introduces a transitive cyclic dependency, which nix is not happy with and it makes maintaining this flake difficult. Instead, in the CI run, we simply build and add k to the PATH temporarily when running the integration tests